Firestore/Source/Core/FSTFirestoreClient.mm

```diff
@@ -252,7 +252,7 @@ - (void)removeListener:(FSTQueryListener *)listener {
 - (void)writeMutations:(NSArray<FSTMutation *> *)mutations
             completion:(nullable FSTVoidErrorBlock)completion {
   [self.workerDispatchQueue dispatchAsync:^{
-    if (mutations.count == 0) {
+    if (completion && mutations.count == 0) {
```

Comment:
This isn't right.

The top-level action here only depends on whether or not any mutations we supplied.

Inside the block we should test if completion has been passed, (similar to the way the action on the `userDispatchQueue` is guarded there.

Reply:
Oops, fixed